test/integration/_shared.js

```diff
@@ -33,6 +33,30 @@ exports.python = {
         await expect(result.trim()).toBe('OK');
     },
 
+    error: ({ expect }, baseURL) => async ({ page }) => {
+        // Test that when the worker throws an error, the page does not crash and the
+        // error is reported to the console.
+        const logs = [];
+        page.on('console', msg => logs.push(msg.text()));
+
+        // GIVEN a page that loads a worker that throws a known error (NameError)
+        await page.goto(`${baseURL}/worker-error.html`);
+        await page.waitForSelector('html.worker.ready');
+        const result = await page.evaluate(() => document.body.innerText);
+
+        // EXPECT the script that runs on the page main thread without errors
+        // to have run successfully
+        await expect(result.trim()).toBe('OK');
+
+        // EXPECT the worker to have thrown a NameError on the console
+        const logText = logs.join(',')
```

Comment:
I would probably `logs.map(({data}) => data.message)` here or something similar, as the listener receives the `error` as it is (but also please check there is something in `logs`)